controllers/deployment.go

```diff
@@ -135,7 +135,20 @@ func (r *DeploymentReconciler) Reconcile(ctx context.Context, application *skipe
 			}
 		}
 
-		container.Resources = application.Spec.Resources
+		container.Resources.Limits = application.LimitToCoreResourceList(application.Spec.Resources.Limits)
+		container.Resources.Requests = application.RequestToCoreResourceList(application.Spec.Resources.Requests)
```

Comment:
We probably need a nil check here. `application.Spec.Resources` is set to required, but there might be some resources on the cluster that have not defined this today which will make this run into a nil pointer exception.